Jenkinsfile

```diff
@@ -10,84 +10,62 @@
-        passwordVariable: "BUILD_SERVER_PASSWORD"
-    )]) {
-        buildServer = [
-            name         : lib.Constants.DEFAULT_PAX_PACKAGING_SSH_HOST,
-            host         : lib.Constants.DEFAULT_PAX_PACKAGING_SSH_HOST,
-            port         : lib.Constants.DEFAULT_PAX_PACKAGING_SSH_PORT.toInteger(),
-            user         : BUILD_SERVER_USERNAME,
-            password     : BUILD_SERVER_PASSWORD,
-            allowAnyHosts: true
-        ]
-    }
-
     pipeline.admins.add("dnikolaev", "sgrady")
 
-    pipeline.setup()
-
-    pipeline.createStage(
-        name: 'Deploy source',
-        stage: {
-            sh "tar --exclude='./source.tar' -cf source.tar *"
-            sshCommand remote: buildServer, command: "rm -rf ${buildDir} && mkdir -p \$_"
-            sshPut remote: buildServer, from: "source.tar", into: buildDir
-            sshCommand remote: buildServer, command: \
-                """
-                cd ${buildDir} &&
-                pax -r -x tar -o to=IBM-1047 -f source.tar
-                """
+    pipeline.setup(
+        packageName: 'org.zowe.zss',
+        github: [
+            email                      : lib.Constants.DEFAULT_GITHUB_ROBOT_EMAIL,
+            usernamePasswordCredential : lib.Constants.DEFAULT_GITHUB_ROBOT_CREDENTIAL,
+        ],
+        artifactory: [
+            url                        : lib.Constants.DEFAULT_ARTIFACTORY_URL,
+            usernamePasswordCredential : lib.Constants.DEFAULT_ARTIFACTORY_ROBOT_CREDENTIAL,
+        ],
+        pax: [
+            sshHost                    : lib.Constants.DEFAULT_PAX_PACKAGING_SSH_HOST,
+            sshPort                    : lib.Constants.DEFAULT_PAX_PACKAGING_SSH_PORT,
+            sshCredential              : lib.Constants.DEFAULT_PAX_PACKAGING_SSH_CREDENTIAL,
+            remoteWorkspace            : lib.Constants.DEFAULT_PAX_PACKAGING_REMOTE_WORKSPACE,
+        ],
+        extraInit: {
+            def version = [
+                sh(script: "cat build/version.properties | grep PRODUCT_MAJOR_VERSION | awk -F= '{print \$2};'", returnStdout: true).trim(),
+                sh(script: "cat build/version.properties | grep PRODUCT_MINOR_VERSION | awk -F= '{print \$2};'", returnStdout: true).trim(),
+                sh(script: "cat build/version.properties | grep PRODUCT_REVISION | awk -F= '{print \$2};'", returnStdout: true).trim(),
+            ].join('.')
+            pipeline.setVersion(version)
+            echo "Current version is v${version}"
         }
     )
 
     pipeline.build(
         operation: {
-            sshCommand remote: buildServer, command: \
-                """
-                export JAVA_HOME=${JAVA_HOME} &&
-                export ANT_HOME=${ANT_HOME} &&
-                export STEPLIB=${XLC_DATASET} &&
-                cd ${buildDir}/build &&
-                ${ANT_HOME}/bin/ant
-                """
+            echo "Build will happen in pre-packaging"
```

Comment:
It's a bit confusing for me to have a build stage in pre-packaging, but as soon as we have binaries to package without conversion, that's a better option.

Reply:
Yes I completely agree with you. Because we cannot actually build in container. Building in pre-packaging is not ideal from this point of view, the encapsulated packaging stage has a lot built-in (like encoding conversion, temp folder handling, etc).

 So it's up to you if you want to choose this implementation, or stick with your way. I'm all ok with it.